pac_man/pacman.py

```diff
@@ -0,0 +1,141 @@
+
+
+class Board:
+    def __init__(self, board_file_name):
+        self.agents = []
+        board_nodes, board_outline = Board.construct_board(board_file_name)
+        self.board_nodes = board_nodes
+        self.board_outline = board_outline
+
+    @staticmethod
+    def construct_board(board_file_name):
+        board_outline = []
+        with open(board_file_name) as f:
+            for line in f:
+                board_outline.append([int(char) for char in line.strip()])
+        board_outline = np.array(board_outline)
+        board_row_nb, board_column_nb = board_outline.shape
+
+        board_nodes = {}
+        # Create nodes
+        for board_row_idx in range(board_row_nb):
+            for board_column_idx in range(board_column_nb):
+                if(board_outline[board_row_idx, board_column_idx] == 1):
+                    new_position = (board_row_idx, board_column_idx)
+                    new_node = Node(new_position)
+                    board_nodes[new_position] = new_node
+        # Associate nodes to children
+        for position, current_node in board_nodes.items():
+            children = []
+            row = position[0]
+            col = position[1]
+            if (row > 0 and (row - 1, col) in board_nodes.keys()):
+                children.append(board_nodes[(row - 1, col)])
+            if (row < board_row_nb and (row + 1, col) in board_nodes.keys()):
+                children.append(board_nodes[(row + 1, col)])
+            if (col > 0 and (row, col - 1) in board_nodes.keys()):
+                children.append(board_nodes[(row, col - 1)])
+            if (col < board_column_nb and
+                    (row, col + 1) in board_nodes.keys()):
+                children.append(board_nodes[(row, col + 1)])
+            current_node.children_nodes = children
+        return board_nodes, board_outline
+
+    def add_agent(self, agent):
+        target_position = agent.current_node.position
+        if (target_position in self.board_nodes):
+            target_node = self.board_nodes[agent.current_node.position]
+            self.agents.append(agent)
+        else:
+            raise OutsideOfLegalPath("""Cannot add agent
+                to invalid board position""")
+
+    def compute_board(self):
+        current_board = self.board_outline.copy()
+        for agent in self.agents:
+            agent_row = agent.current_node.position[0]
+            agent_col = agent.current_node.position[1]
+            if (isinstance(agent, PacMan)):
+                current_board[agent_row, agent_col] = 2
+            if (isinstance(agent, Ghost)):
+                current_board[agent_row, agent_col] = 3
+        return current_board
+
+    def draw_board(self, speed):
+        current_board = self.compute_board()
+        plt.figure(1)
+        plt.matshow(current_board, fignum=1)
+        plt.pause(speed)
+
+    def __str__(self):
+        current_board = self.compute_board()
+        return current_board.__str__()

```

Comment:
If I am not mistaken, you are not suppose to access hidden methods with underscores.
The correct way to call `current_board.__str__()` is simply through `str(current_board)`, it should be working.